death-causes-app/src/components/Calculations/CalculationService.ts

```diff
@@ -0,0 +1,227 @@
+import { propTypes } from "react-bootstrap/esm/Image";
+import { FactorAnswers } from "../../models/Factors";
+import { ProbabilityKeyValue } from "../../models/ProbabilityKeyValue";
+import DeathCause from "../database/Deathcause";
+import { ProbabilityOfDeathCause, ProbabilitiesOfAllDeathCauses } from "../database/ProbabilityResult";
+import { RiskFactorGroup } from "../database/RickFactorGroup";
+import { MinimumRiskRatios } from "../database/RiskRatioTable";
+import { RiskRatioTableCellInterface } from "../database/RiskRatioTableCell/RiskRatioTableCellInterface";
+import { DataRow } from "../PlottingData";
+import { SurvivalCurveData } from "./SurvivalCurveData";
+
+interface MinimumFactorInputs {
+    [key: string]: RiskRatioTableCellInterface
+}
+
+interface UStar {
+    propForDeathcause: number,
+    minProbForDeathcause: number,
+    minFactorIntervals: MinimumFactorInputs,
+    minFactorValues: FactorAnswers;
+}
+
+export class RiskRatioCalculationService {
+    private readonly MAX_AGE = 120;
+
+    calculateSurvivalCurve(submittedFactorAnswers: FactorAnswers, deathcauses: DeathCause[]): SurvivalCurveData[] {
+        let probabilitiesPerDeathCause = this.calculateProbabilitiesOfDeathCauses(submittedFactorAnswers, deathcauses)
+        let deathCauseProbabilities = probabilitiesPerDeathCause.probabilitiesOfAllDeathCauses.map(x => x.probabiltiesOfDeathCause)
+
+        let totalProbabilityOfNotDying: number[] = Array.from({ length: probabilitiesPerDeathCause.ages.length })
+        totalProbabilityOfNotDying = totalProbabilityOfNotDying.map((_, i) => deathCauseProbabilities.map(val => val[i]).reduce((sum, x) => sum + x, 0))
+            .map(x => 1 - x)
+
+        let res: SurvivalCurveData[] = []
+        let survivalCurve = Array.from({ length: totalProbabilityOfNotDying.length - 1 }, () => 1)
+        for (let i = 1; i < totalProbabilityOfNotDying.length; i++) {
+            survivalCurve[i] = survivalCurve[i - 1] * totalProbabilityOfNotDying[i]
+            res.push({age: probabilitiesPerDeathCause.ages[i], prob: survivalCurve[i]})
+        }
+
+        return res
+    }
+
+    calculateProbabilitiesOfDeathCauses(submittedFactorAnswers: FactorAnswers, deathcauses: DeathCause[]): ProbabilitiesOfAllDeathCauses {
```

Comment:
Både her og i inner_causes mangles der at gange med sandsynligheden for at være i live til hver alder